.github/workflows/publish.yml

```diff
@@ -1,6 +1,12 @@
 name: Publish to PyPI
 
 on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
is this correct? We want to publish to PyPI whenever there's a push or a pull request for the main branch?

Reply:
This will ensure the tarball tests here will run. We have na if clause below that will publish only when there is a tag.